packages/core/lib/commands/opcode/run.js

```diff
@@ -31,8 +33,8 @@ module.exports = async function (options) {
 
   const indexLength = (opcodes.length + "").length;
 
-  opcodes.forEach((opcode, index) => {
-    let strIndex = index + ":";
+  opcodes.forEach(opcode => {
+    let strIndex = Conversion.toHexString(opcode.pc) + ":";
 
     while (strIndex.length < indexLength + 1) {
```

Comment:
The thing I think most needs changing here is this loop for padding the displayed value.  It left-pads with spaces to get things to a length of `indexLength + 1`.  (I guess it was written back before `str.padStart()` existed.)  But, I think that's not really what we want, for two reasons:

1. (The more important reason) The variable `indexLength` is based on the length of the largest *index* in decimal; but here we're no longer going based on indices but rather based on PC values.  So the length to pad to should be based on the largest PC value, not the largest index.

2. (The less important reason)  I think the following looks a bit awkward:

```
  0xff: POP
0x0100: POP
```

Right?  Instead, it should probably be:

```
0x00ff: POP
0x0100: POP
```

So, instead of left-padding with spaces to the left of the `0x`, it would make more sense to left-pad with 0s to the right of the `0x`.

Of course the good news is that, since you're already using `Codec.Conversion.toHexString`, you can use its functionality for that; it allows a second argument which is a minimum length to use.  Note however that this length is given in bytes, rather than hex digits, so make sure you account for that if you want to do it that way.  (Or you could always do it manually, obviously, but that seems more awkward to me...)

Reply:
Done! Maybe the naming is a little strange, if there is a better option I will change 🙂

The idea is that we look at the last PC, find out its length in bytes and make padding based on it.